eng/illink.targets

```diff
@@ -64,6 +75,36 @@
     </BinPlaceTargetFramework>
   </ItemGroup>
 
+  <Target Name="_EmbedILLinkSubstitutionsXmls"
+          Condition="'$(ILLinkTrimAssembly)' == 'true'"
+          DependsOnTargets="_CombineILLinkSubstitutionsXmls">
+
+    <ItemGroup Condition="'$(ILLinkSubstitutionsXml)' != ''">
+      <EmbeddedResource Include="$(ILLinkSubstitutionsXml)">
+        <LogicalName>ILLink.Substitutions.xml</LogicalName>
+      </EmbeddedResource>
+    </ItemGroup>
+
+  </Target>
+
+  <UsingTask TaskName="CombineLinkerXmlFiles" AssemblyFile="$(ILLinkTasksPath)" />
+  <Target Name="_CombineILLinkSubstitutionsXmls"
+          Condition="'$(ILLinkSubstitutionsXml)' == '' and '@(ILLinkSubstitutionsXml)' != ''"
+          Inputs="@(ILLinkSubstitutionsXml)"
+          Outputs="$(ILLinkSubstitutionsXmlIntermediatePath)">
+    <PropertyGroup>
+      <ILLinkSubstitutionsXml>$(ILLinkSubstitutionsXmlIntermediatePath)</ILLinkSubstitutionsXml>
+    </PropertyGroup>
+
+    <CombineLinkerXmlFiles Condition="'@(ILLinkSubstitutionsXml)' != ''"
```

Comment:
NIT: no need to check again if the item is empty as you checked on the target already

Reply:
Good call. Removing the condition here.